### DISCUSSION TITLE
Offline Relay Climate Control

### DISCUSSION THREAD

**Discussion by andersagerskov** [Blackymas/NSPanel_HA_Blueprint#263] [Ideas]
Thanks for a great project! Huge potential with this!

I am running NSPanels and use them to also control my floor heating - simple 230v cabled actuators.

Would it be possible to have an "emergency" code piece running directly in the NSPanel that would allow the heating (opening/closing of physical relay) based on temperature to continue working even if WiFi/HA is down?

I know the screen will not be responsive, changes cannot be done to settings, but it would make the solution close to perfect if this simple relay control could continue regardless of connection. 

/Anders

**Comment by Blackymas** [comment]
Hello and thank you.

I also control my underfloor heating with it.

In the next version 3.1.0 is a fallback function in it.

If the NSPanel loses the connection to WiFi or HA is immediately activated the fallback and ud can switch with the hardware buttons the releay locally on the NSPanel.

**Reply by andersagerskov** [reply]
Sounds like it will cover my needs! Look forward to testing the next version.

Thank you for the quick reply!

/Anders

**Comment by e120guru** [comment]
Will this fallback function by so that it is manually controllable? Or will it function as a ”dumb” thermostat, and switch on/off based on the setpoint?

**Comment by Blackymas** [comment]
manually controllable

**Reply by e120guru** [reply]
Thanks, I’m going to write in another Thread what I’m looking at achieving… I have not delat much with specifically ESPhome based things, so I don’t know how much freedom there is, but anyway - I would be open to helping out with some implementation once I have my physical units. (Been working for quite some time with Embedded software development)

**Reply by e120guru** [reply]
I changed my mind, I'm writing in a separate post in this thread instead as it's very related to this topic.

**Comment by e120guru** [comment]
I'm awaiting delivery of NSpanel to play with. But, here's what I'm in the process now of doing:

First, a little bit of background: I live in a 2 story house with water based floor heating. The house has 4 water circuits on the bottom floor, and 5 water circuits on the second floor. These are controlled through 24VAC thermostats - this is a quite typical way here in Sweden. This means that I have 24VAC going to the thermostat, and 24VAC going back that the thermostat switches on/off to control the wax motor valve control to control the flow in the floor heating.

I'm making a PCB that will replace the mains part of the NSpanel, meaning that instead of 230V going in to power, and having relayed outputs, this PCB will take the 24VAC and feed the panel with 5V, and have SSRs to control 24VAC going out.

But, since the floor heating is the only means of heating we have in the house, I do not want to risk in getting stuck in either full open or fully closed if Home Assistant dies. Since it cound become very warm or very cold in the house in that case (at least in some rooms - as the heat pump is adjusted so that the coldest rooms usually have the valve fully open.
This means that for this ti be a reliable replacement without the need to worry - I would like the unit to act as a stand-alone thermostat if/when it looses connection with HA (or rather - if automations in HA stops).

I'm happy to help in implementing this - but I don't know how flexible the underlaying platform actually is in terms of doing these kind of things, or is all logic running in HA?

**Reply by deejaybeam** [reply]
YES - all logic is running in HA.
The NSPanel has very limited resources.
There is a climate/thermostat component for ESPHome that can be used for your case, but this would "waste" resources on the device itself in many use-cases.

You are afraid of outage of hassio? Invest in good hardware (e.g. no raspberry with SD-Card), take care about remote-management (vpn, ...).
What if the display or the motor valve, or your self-made PCB, or the heat-pump, etc. fails?

But anyway this is an open-source project and it´s really easy to add a climate component to the esphome - config by your own:
https://esphome.io/components/climate/index.html
https://esphome.io/components/climate/bang_bang.html

**Reply by andersagerskov** [reply]
I am in the same situation as @e120guru, with 10 separate channels with water based floor heating. To me it would also be optimal if the fallback would be able to run locally in the NSPanel, with just the last values as fixed values until the HA was available again. 

The other components used in my setup are very reliant, cabled connections to thermostats, cabled actuators which are extremely simple Normal-Close components, so I have to second the fact the possibility of WiFI/HA failure is much bigger than the rest of the heating setup. And knowing good old Murphy a failure/error would always happen when you are away on weekend during the coldest day of the year.

Maybe an option to build this feature into the image would be possible, and then for those of us who would like this "backup" would have to sacrifice some other features that would not have enough resources. Not sure if possible?

/Anders

**Comment by e120guru** [comment]
Yes, as Anders mentioned - the WiFi connection itself or HA instance are the most likelly failures, and also the once potentially causing the most issues.

Most part of the heating system are designed for 15-20 years service life. And there are som redundancy for the most likelly failures that causes the largest problems - for instance If the compressor dies, there’s backup heating elements. The thermostats are wired, etc.
If a valve fails, that affects only one room and not the entire house.
The same with if a for instance the relay, the voltage regulator or the NSpanel itself would fail, then it also would affect only one room.
But if HA crashes it would affect the entire house. 

Of course the pump could short and trip the fuses, so there is always a risk of a single point of failure for the entire system. Just don’t want that point to be the same point as the one that’s most likelly to fail ;)

I’ll have a look at trying to get some local control, I have never placed with ESPhome, so not too familiar with how it works, but will have a look after I get my devices.

**Comment by edwardtfn** [comment]
A local control thermostat is coming to life... ;)
#917 
I still have to test and fix some bugs, but first (huge) step was made.

**Comment by edwardtfn** [comment]
I'm looking for early adopters to test the new embedded climate control before the release: https://github.com/Blackymas/NSPanel_HA_Blueprint/discussions/995

**Comment by edwardtfn** [comment]
I'm closing this discussion as we have released v4.0 (#1057).
Please add a feature request for the missing things.

And thanks a lot for your support.
